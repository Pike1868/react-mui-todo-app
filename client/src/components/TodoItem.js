import React, { useState } from "react";
import { Box, Button, ListItem } from "@mui/material";
import {
  Edit as EditIcon,
  Delete as DeleteIcon,
  CheckBoxOutlineBlankOutlined as BlankCheckBoxIcon,
} from "@mui/icons-material";
import Modal from "./Modal";

const TodoItem = ({ task, getData, deleteItem }) => {
  const [showModal, setShowModal] = useState(false);

  return (
    <ListItem
      className="list-item"
      sx={{
        border: "2px solid #c5c5c5",
        borderRadius: "1rem",
        marginBottom: "1rem",
        display: "flex",
        justifyContent: "space-between",
        padding: "0.5rem",
        backgroundColor: "#fff",
      }}
    >
      <Box className="info-container">
        <BlankCheckBoxIcon
          onClick={() => {
          }}
        />
        <p className="task-title">{task.title}</p>
        <strong>{task.progress}% Complete</strong>
      </Box>
      <Box
        sx={{
          display: "flex",
          flexDirection: "column",
          justifyContent: "space-around",
          width: "25%",
          height: "100%",
        }}
      >
        <Button
          size="small"
          variant="contained"
          startIcon={<EditIcon />}
          color="secondary"
          sx={{ borderRadius: 5, fontSize: "0.8rem", marginBottom: "0.5rem" }}
          onClick={() => setShowModal(true)}
        >
          EDIT
        </Button>
        <Button
          size="small"
          variant="contained"
          startIcon={<DeleteIcon />}
          color="warning"
          sx={{ borderRadius: 5, fontSize: "0.8rem" }}
          onClick={() => {
            deleteItem(task.id);
          }}
        >
          Delete
        </Button>
      </Box>
      {showModal && (
        <Modal
          mode={"edit"}
          setShowModal={setShowModal}
          getData={getData}
          task={task}
        />
      )}
    </ListItem>
  );
};

export default TodoItem;
