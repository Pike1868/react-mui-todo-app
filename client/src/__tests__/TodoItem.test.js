import React from "react";
import TodoItem from "../components/TodoItem";
import { render, screen } from "@testing-library/react";

//Mock Task
const mockTask = {
  id: "0",
  user_email: "todo@test.com",
  title: "First Todo",
  progress: 10,
  date: "Tue Dec 26 2023 15:36:11 GMT-0500 (Eastern Standard Time)",
};

//Smoke Test
it("renders without crashing", () => {
  render(<TodoItem task={mockTask} />);
});

//Snapshot Test
it("matches snapshot", () => {
  const { asFragment } = render(<TodoItem task={mockTask} />);
  expect(asFragment()).toMatchSnapshot();
});
