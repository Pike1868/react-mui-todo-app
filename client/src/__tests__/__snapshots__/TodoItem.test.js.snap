// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding list-item css-jgp9yu-MuiListItem-root"
  >
    <div
      class="info-container MuiBox-root css-0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CheckBoxOutlineBlankOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <p
        class="task-title"
      >
        First Todo
      </p>
      <strong>
        10% Complete
      </strong>
    </div>
    <div
      class="MuiBox-root css-plnpdn"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall css-76gxe9-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="EditIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </span>
        EDIT
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedWarning MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedWarning MuiButton-sizeSmall MuiButton-containedSizeSmall css-m8gqhu-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DeleteIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        Delete
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </li>
</DocumentFragment>
`;
